distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1683,13 +1718,97 @@ private CometDClient setupCometDClient(List<String> channels) throws Exception {
         String cometDEndpointUrl = COMETD_ENDPOINT.getUrl();
         CometDClient cometDClient = new CometDClient(cometDEndpointUrl);
         cometDClient.start();
-        channels.stream()
-                .forEach(cometDClient::subscribe);
+        channels.forEach(cometDClient::subscribe);
         return cometDClient;
     }
 
+    private void setupParallelInterruptDownloadTest(String metacardId, String filename,
+            String resourceData) throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(50), 0));
+    }
```

Comment:
Lines 1727-1740 could potentially be replaced by a DownloadHandle object of the class I added to TestFederation in my PR: https://github.com/codice/ddf/pull/1009